src/Http/Routing/src/Internal/LinkGenerationDecisionTree.cs

```diff
@@ -139,6 +169,12 @@ private class OutboundMatchResultComparer : IComparer<OutboundMatchResult>
 
             public int Compare(OutboundMatchResult x, OutboundMatchResult y)
             {
+                if (x.Quality != y.Quality)
+                {
+                    // Reversed because higher is better
+                    return y.Quality.CompareTo(x.Quality);
+                }
```

Comment:
I made this change while pursuing option 2, I don't think it's needed for anything so I could revert it once we decide what to do.